molo/core/utils.py

```diff
@@ -58,19 +59,42 @@ def generate_slug(text, tail_number=0):
         return slug
 
 
-def replace_media_file(file):
-    '''Replace the Current Media Folder.'''
-    media_parent_directory = os.path.dirname(settings.MEDIA_ROOT)
-    zip_file_reference = os.path.join(media_parent_directory, 'new_media.zip')
-
-    if os.path.isdir(settings.MEDIA_ROOT):
-        shutil.rmtree(settings.MEDIA_ROOT)
-
-    with open(zip_file_reference, 'wb+') as destination:
-        for chunk in file.chunks():
-            destination.write(chunk)
+def update_media_file(file):
+    '''
+    Update the Current Media Folder.
 
-    with zipfile.ZipFile(zip_file_reference, 'r') as z:
-        z.extractall(media_parent_directory)
-
-    os.remove(zip_file_reference)
+    Returns list of files copied accross or
+    raises an exception.
+    '''
+    media_parent_directory = os.path.dirname(settings.MEDIA_ROOT)
+    temp_dir_path = os.path.join(media_parent_directory, 'temp_copy_dir')
```

Comment:
Consider using python's [Temp File](https://docs.python.org/2/library/tempfile.html)
It should make managing the temp file easier